frontend/packages/dev-console/src/components/import/section/ResourceSection.tsx

```diff
@@ -31,14 +41,87 @@ const createHelpText = (k8sModel: K8sKind, helpText: string) => {
 const ResourceSection: React.FC<ResourceSectionProps> = ({ flags }) => {
   const { t } = useTranslation();
   const [field] = useField<Resources[]>('resourceTypesNotValid');
+  const fieldName = 'resources';
+  const {
+    values: { formType },
+    setFieldValue,
+  } = useFormikContext<FormikValues>();
   const invalidTypes = field.value || [];
+
   const knativeServiceAccess = useAccessReview({
     group: ServiceModel.apiGroup,
     resource: ServiceModel.plural,
     namespace: getActiveNamespace(),
     verb: 'create',
   });
+  const canIncludeKnative =
+    !invalidTypes.includes(Resources.KnativeService) &&
+    flags[FLAG_KNATIVE_SERVING_SERVICE] &&
+    knativeServiceAccess;
+
+  const defaultResourceType = canIncludeKnative ? Resources.KnativeService : Resources.Kubernetes;
+
+  const [
+    preferredResourceType,
+    setPreferredResourceType,
+    preferredResourceTypeLoaded,
+  ] = useUserSettings<string>(PREFERRED_RESOURCE_TYPE_USER_SETTING_KEY, defaultResourceType, true);
+
+  const [resourceType, setResourceType, resourceTypeLoaded] = useUserSettingsCompatibility<string>(
+    LAST_RESOURCE_TYPE_STORAGE_KEY,
+    PREFERRED_RESOURCE_TYPE_LOCAL_STORAGE_KEY,
+    defaultResourceType,
+    true,
+  );
+
+  React.useEffect(() => {
+    if (formType !== 'edit' && !flags[FLAG_KNATIVE_SERVING_SERVICE]) {
+      if (resourceTypeLoaded && resourceType === Resources.KnativeService) {
```

Comment:
Shouldnt this be:
```suggestion
      if (preferredResourceTypeLoaded  && resourceTypeLoaded && preferredResourceType === 'latest' && resourceType === Resources.KnativeService) {
```

Reply:
Ah!! correct. My bad. updated it. Thanks!!